### ISSUE TITLE
encountered an intermittent signal6 during rolling restart

### ISSUE THREAD

**Issue by mchang31415** [YugaByte/yugabyte-db#37]
```
Core was generated by `/home/yugabyte/tserver/bin/../lib/unwrapped/yb-tserver --flagfile /home/yugabyt'.
Program terminated with signal 6, Aborted.
#0  0x00007efc461f3067 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
54      ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) where
#0  0x00007efc461f3067 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007efc461f446a in __GI_abort () at abort.c:89
#2  0x00007efc461ec1b6 in __assert_fail_base (fmt=0x7efc46325118 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n",
    assertion=assertion@entry=0x7efc4b7b5030 "(\"libev: internal timer heap corruption\", ANHE_w (timers [active]) == (WT)w)", file=file@entry=0x7efc4b7b48e6 "ev.c", line=line@entry=3776,
    function=function@entry=0x7efc4b7b5390 <__PRETTY_FUNCTION__.6920> "ev_timer_stop") at assert.c:92
#3  0x00007efc461ec262 in __GI___assert_fail (assertion=0x7efc4b7b5030 "(\"libev: internal timer heap corruption\", ANHE_w (timers [active]) == (WT)w)", file=0x7efc4b7b48e6 "ev.c", line=3776,
    function=0x7efc4b7b5390 <__PRETTY_FUNCTION__.6920> "ev_timer_stop") at assert.c:101
#4  0x00007efc4b7b0e86 in ev_timer_stop (loop=0x21ba400, w=0x2cdb6d520) at ev.c:3776
#5  0x00007efc4bc24f3a in stop (this=0x2cdb6d520) at /n/jenkins/thirdparty/yugabyte-thirdparty-2018-01-09T08_37_15/thirdparty/installed/common/include/ev++.h:638
#6  ~timer (this=0x2cdb6d520, __in_chrg=<optimized out>) at /n/jenkins/thirdparty/yugabyte-thirdparty-2018-01-09T08_37_15/thirdparty/installed/common/include/ev++.h:638
#7  yb::rpc::Connection::~Connection (this=0x2cdb6d410, __in_chrg=<optimized out>) at ../../../../../src/yb/rpc/connection.cc:89
#8  0x00007efc4bc2ad62 in _M_release (this=0x2cdb6d400) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/bits/shared_ptr_base.h:150
#9  ~__shared_count (this=0x427c6080, __in_chrg=<optimized out>) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/bits/shared_ptr_base.h:659
#10 ~__shared_ptr (this=0x427c6078, __in_chrg=<optimized out>) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/bits/shared_ptr_base.h:925
#11 ~shared_ptr (this=0x427c6078, __in_chrg=<optimized out>) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/bits/shared_ptr.h:93
#12 yb::rpc::InboundCall::~InboundCall (this=0x427c6010, __in_chrg=<optimized out>) at ../../../../../src/yb/rpc/inbound_call.cc:90
#13 0x00007efc51eec3e6 in ~QueueableInboundCall (this=0x427c6010, __in_chrg=<optimized out>) at ../../../../../../../src/yb/rpc/rpc_with_queue.h:29
#14 yb::redisserver::RedisInboundCall::~RedisInboundCall (this=0x427c6010, __in_chrg=<optimized out>) at ../../../../../../../src/yb/yql/redis/redisserver/redis_rpc.cc:130
#15 0x00000000004113a6 in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release (this=0x427c6000)
    at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/bits/shared_ptr_base.h:150
#16 0x00007efc51efe523 in ~__shared_count (this=0x138a3588, __in_chrg=<optimized out>) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/bits/shared_ptr_base.h:659
#17 ~__shared_ptr (this=0x138a3580, __in_chrg=<optimized out>) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/bits/shared_ptr_base.h:925
#18 ~shared_ptr (this=0x138a3580, __in_chrg=<optimized out>) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/bits/shared_ptr.h:93
#19 ~BatchContext (this=0x138a3560, __in_chrg=<optimized out>) at ../../../../../../../src/yb/yql/redis/redisserver/redis_service.cc:531
#20 DeleteInternal (x=0x138a3560) at ../../../../../../../src/yb/gutil/ref_counted.h:166
#21 Destruct (x=0x138a3560) at ../../../../../../../src/yb/gutil/ref_counted.h:129
#22 yb::RefCountedThreadSafe<yb::redisserver::(anonymous namespace)::BatchContext, yb::DefaultRefCountedThreadSafeTraits<yb::redisserver::(anonymous namespace)::BatchContext> >::Release (
    this=this@entry=0x138a3560) at ../../../../../../../src/yb/gutil/ref_counted.h:157
#23 0x00007efc51f03964 in ~scoped_refptr (this=<synthetic pointer>, __in_chrg=<optimized out>) at ../../../../../../../src/yb/gutil/ref_counted.h:280
#24 operator() (status=..., this=<optimized out>) at ../../../../../../../src/yb/yql/redis/redisserver/redis_service.cc:379
#25 boost::detail::function::void_function_obj_invoker1<yb::redisserver::(anonymous namespace)::Block::BlockCallback, void, yb::Status const&>::invoke (function_obj_ptr=..., a0=...)
    at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/include/boost/function/function_template.hpp:159
#26 0x00007efc4f903796 in operator() (a0=..., this=0x33094d638) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/include/boost/function/function_template.hpp:760
#27 yb::client::internal::Batcher::CheckForFinishedFlush (this=0x33094d600) at ../../../../../src/yb/client/batcher.cc:192
#28 0x00007efc4f8fa642 in yb::client::internal::AsyncRpc::Finished (this=0xd054790, status=...) at ../../../../../src/yb/client/async_rpc.cc:136
#29 0x00007efc4bc376bf in operator() (this=0x2f541b910) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/functional:2267
#30 yb::rpc::OutboundCall::CallCallback (this=this@entry=0x2f541b890) at ../../../../../src/yb/rpc/outbound_call.cc:327
#31 0x00007efc4bc386b5 in yb::rpc::OutboundCall::SetResponse(yb::rpc::CallResponse&&) (this=this@entry=0x2f541b890,
    resp=resp@entry=<unknown type in /home/yugabyte/tserver/bin/../lib/yb/libyrpc.so, CU 0x2aaf2f, DIE 0x349f1d>) at ../../../../../src/yb/rpc/outbound_call.cc:366
#32 0x00007efc4bc2788c in yb::rpc::Connection::HandleCallResponse (this=0x5bc26a10, call_data=...) at ../../../../../src/yb/rpc/connection.cc:409
#33 0x00007efc4bc6b50e in yb::rpc::YBConnectionContext::HandleCall (this=this@entry=0x56b3e870, connection=..., call_data=...) at ../../../../../src/yb/rpc/yb_rpc.cc:166
#34 0x00007efc4bc6b83e in yb::rpc::YBConnectionContext::ProcessCalls (this=0x56b3e870, connection=..., slice=..., consumed=0x7efc0e8b06b0) at ../../../../../src/yb/rpc/yb_rpc.cc:142
#35 0x00007efc4bc2261c in yb::rpc::Connection::TryProcessCalls (this=this@entry=0x5bc26a10) at ../../../../../src/yb/rpc/connection.cc:378
#36 0x00007efc4bc22906 in yb::rpc::Connection::ReadHandler (this=this@entry=0x5bc26a10) at ../../../../../src/yb/rpc/connection.cc:330
#37 0x00007efc4bc23b6d in yb::rpc::Connection::Handler (this=0x5bc26a10, watcher=..., revents=1) at ../../../../../src/yb/rpc/connection.cc:286
#38 0x00007efc4b7af717 in ev_invoke_pending (loop=0x371ff80) at ev.c:3155
#39 0x00007efc4b7b061e in ev_run (loop=0x371ff80, flags=0) at ev.c:3555
#40 0x00007efc4bc449a9 in run (flags=0, this=0x21eb4e0) at /n/jenkins/thirdparty/yugabyte-thirdparty-2018-01-09T08_37_15/thirdparty/installed/common/include/ev++.h:211
#41 yb::rpc::Reactor::RunThread (this=0x21eb4a0) at ../../../../../src/yb/rpc/reactor.cc:424
#42 0x00007efc4a3005c6 in operator() (this=0x2e341c8) at /n/jenkins/linuxbrew/linuxbrew_2018-01-09T08_28_02/Cellar/gcc/5.5.0/include/c++/5.5.0/functional:2267
#43 yb::Thread::SuperviseThread (arg=<optimized out>) at ../../../../../src/yb/util/thread.cc:602
#44 0x00007efc46564694 in start_thread (arg=0x7efc0e8b1700) at pthread_create.c:333
#45 0x00007efc462a63cd in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

**Comment by kmuthukk** [comment]
CC: @spolitov , @amitanandaiyer (for initial look).

**Comment by mbautin** [comment]
Fixed in 6c4d38b0fa06cb9bdb4a58e994c5fe5e8652e3b4.
